### PR TITLE
Use the same environment as the docker image mdernst/ubuntu-for-cf-jdk8

### PR DESCRIPTION
**Pull request by zhangjiangqige**
Currently the build is failing because Bitbucket has deprecated the support of TLS 1.0 and now only supports 1.2, which seemingly requires a higher version of Python that's not installed in the default Ubuntu Trusty (version 14) environment.

So I think it may be a good idea to switch to Ubuntu Xenial (version 16), as in other repos which are using the docker image `mdernst/ubuntu-for-cf-*` and openjdk 8 (and oraclejdk8 is actually not installed/supported in Xenial somehow).

### CODE REVIEW

**Comment by zhangjiangqige** [comment]
Merging both https://github.com/eisop/checker-framework.demos/pull/1 and this PR should make the build successful like [this](https://travis-ci.com/zhangjiangqige/checker-framework.demos/builds/114157505).

**Comment by wmdietl** [comment]
Why is this not necessary in the typetools version? There is nothing Java 9 specific about this change, so I don't understand why this is necessary. Do you?

**Comment by zhangjiangqige** [comment]
> Why is this not necessary in the typetools version? There is nothing Java 9 specific about this change, so I don't understand why this is necessary. Do you?

Because the typetools version of annotation-tools doesn't seem to require jsr308-langtools, whereas the eisop version of annotation-tools requires that.

This can be seen from the difference of the build files of these two repos: [typetools](https://github.com/typetools/annotation-tools/blob/032bcd17e8b1b64569990fc349798b40bf272ee1/.travis-build-without-test.sh#L29) and [eisop](https://github.com/eisop/annotation-tools/blob/4bf7b819a1b378974f89b9ef9ad60cf701f513bc/.travis-build-without-test.sh#L29). There's a section in the eisop version build file cloning jsr308-langtools from Bitbucket, which doesn't exist in the typetools version.

**Comment by wmdietl** [comment]
Ah, yes, that makes sense. However, I think it would be nicer to make this .travis.yml file closer to the environment we use in the other projects, e.g. https://github.com/eisop/checker-framework/blob/master/.travis.yml
Reducing differences would be good.

**Comment by wmdietl** [comment]
(The alternative would be to use the pre-built error-prone javac like the gradle build already does. I'm not sure how many changes would be required to use that everywhere.)

**Comment by zhangjiangqige** [comment]
Now this repo uses the docker image as the other projects. Although it's not using the error-prone javac but it can be built successfully on Traivs.

File: .travis.yml [hunk 1]
```diff
@@ -18,8 +9,65 @@ cache:
...
+     export TRAVIS_REPO_SLUG=$TRAVIS_REPO_SLUG &&
+     export TRAVIS_COMMIT_RANGE=$TRAVIS_COMMIT_RANGE &&
+     export TRAVIS_PULL_REQUEST_BRANCH=${TRAVIS_PULL_REQUEST_BRANCH} &&
+     export TRAVIS_PULL_REQUEST_SLUG=${TRAVIS_PULL_REQUEST_SLUG} &&
+     export TRAVIS_BRANCH=${TRAVIS_BRANCH} &&
+     ./.travis-build.sh"
 
 git:
-  depth: 3
+  depth: 9
```
Comment (wmdietl) [inline .travis.yml:73]: Please add the missing end-of-line here. Notice the red error sign.

**Review by wmdietl — COMMENTED** [review]
Thanks, this looks great!
I have one small comment to address. Afterward, please move the PR to typetools and let's merge it there. This seems like a generally useful change, so let's not merge it in the Java 9 version.

**Comment by zhangjiangqige** [comment]
> Thanks, this looks great!
> I have one small comment to address. Afterward, please move the PR to typetools and let's merge it there. This seems like a generally useful change, so let's not merge it in the Java 9 version.

Just created https://github.com/typetools/checker-framework.demos/pull/7 with an end-of-line. I'll close this one.

**Pull request closed without merging**